### DISCUSSION TITLE
RT3S radio not detected in linux

### DISCUSSION THREAD

**Discussion by detournemint** [hmatuschek/qdmr#176] [Q&A] [answered]
Says "No Matching Radio Found" Anyway to get it to detect?

**Comment by hmatuschek** [comment]
This may depend a bit on how you installed qdmr and the distribution. 

First, can you check if the radio is detected by the kernel. Call `lsusb` an look for a VID/PID `0483:df11`. It should be named something like `STM device in DFU mode`. Also, your device should not be in firmware flash mode. Do not press the side buttons while switching it on.

If you installed the Snap package, you need to allow qdmr raw access to USB devices (see https://dm3mat.darc.de/qdmr/install.html) with 
```
snap connect qdmr:raw-usb
```
to grant the access.

If you installed qdmr by building it from the source code, ensure that the udev rules are installed properly. There should be a file name `99-qdmr.rules` in `/etc/udev/rules.d/`. If it is there try running 
```
sudo udevadm control --reload-rules
```
to reload these rules (although this should have happened automatically).

If you installed one of the pre-build packages for Ubuntu, Arch, OpenSuSE or Gentoo. It should work out of the box. To investigate it further, install (if not already present) `dmrconf` the command line version of qdmr. Then run
``` 
dmrconf --verbose detect
```
and dump the output here.

**Comment by detournemint** [comment]
I'm not finding the `dmrconf` package, i found `dmrconfig` but it doesn't have that command. I'm running ubuntu and i installed qdmr from the software store

**Comment by hmatuschek** [comment, marked as answer]
Ok I see two issues here.,

First, when you install qdmr via the *software store* in Ubuntu, the snap package gets installed by default. This variant should work with the RT3S. However, you need to grant qdmr raw access to the USB devices. This can be done by calling 
```
snap connect qdmr:raw-usb
```
This should do the trick. 

However, even then I see a problem: Your lsusb list you send, did not contain the device. There must be a device listed, that is called "STM device in DFU mode" or something like that. I saw, however, that there was a device called "FT232 Serial (UART) IC". This one and similar chips are used in some cables to connect to HTs. However, not for the Retevis RT3S. There a direct USB connection is used to the MCU within the radio. Are you sure that you are using the correct cable for this device? Does this cable work under windows with the manufacturer CPS?

**Comment by detournemint** [comment]
I was able to get it working using the cable that came with the radio,
Thank to you so much

On Wed, Nov 10, 2021 at 6:43 AM Hannes Matuschek ***@***.***>
wrote:

> Ok I see two issues here.,
>
> First, when you install qdmr via the *software store* in Ubuntu, the snap
> package gets installed by default. This variant should work with the RT3S.
> However, you need to grant qdmr raw access to the USB devices. This can be
> done by calling
>
> snap connect qdmr:raw-usb
>
> This should do the trick.
>
> However, even then I see a problem: Your lsusb list you send, did not
> contain the device. There must be a device listed, that is called "STM
> device in DFU mode" or something like that. I saw, however, that there was
> a device called "FT232 Serial (UART) IC". This a similar chips are used in
> some cables to connect to HTs. However, not for the Retevis RT3S. There a
> direct USB connection is used to the MCU within the radio. Are you sure
> that you are using the correct cable for this device? Does this cable work
> under windows with the manufacturer CPS?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/hmatuschek/qdmr/issues/173#issuecomment-965301859>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFGVOCNIR2QNXU4HHSJZLLULKAKRANCNFSM5HQI4IVQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Gregg Horton
gregghorton.com

**Comment by detournemint** [comment]
Sorry to revist this but I was trying to set this up on a new computer, I used your ppa package but things aren't working again

```
$ dmrconf --verbose detect
Debug in dfu_libusb.cc@39: Try to detect USB DFU interface 483:df11.
ERROR in dfu_libusb.cc@50: Cannot open device 483:df11: Success
ERROR in tyt_interface.cc@11: Cannot open TyTInterface.
Debug in hid_libusb.cc@12: Try to detect USB HID interface 15a2:73.
ERROR in hid_libusb.cc@27: Cannot find USB device 15a2:73.
Debug in usbserial.cc@11: Try to detect USB serial interface 1fc9:94.
ERROR in usbserial.cc@36: No serial port found with 1fc9:94.
Debug in usbserial.cc@11: Try to detect USB serial interface 28e9:18a.
ERROR in usbserial.cc@36: No serial port found with 28e9:18a.
ERROR in radio.cc@471: Cannot connect to radio.
ERROR in detect.cc@17: No compatible radio found: 

$ lsusb
Bus 002 Device 002: ID 0bda:0316 Realtek Semiconductor Corp. USB3.0-CRW
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 004: ID 04ca:7066 Lite-On Technology Corp. Integrated Camera
Bus 001 Device 003: ID 8087:0a2b Intel Corp. 
Bus 001 Device 002: ID 0483:df11 STMicroelectronics STM Device in DFU Mode
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

```

**Reply by detournemint** [reply]
well i didn't add myself to the dialout, that fixed it

**Reply by hmatuschek** [reply]
Yes, with 0.9.3, the udev rules require a membership in `dialout`.
